service/history/transfer_queue_active_task_executor.go

```diff
@@ -85,7 +85,8 @@ func newTransferQueueActiveTaskExecutor(
 	logger log.Logger,
 	metricProvider metrics.Handler,
 	config *configs.Config,
-	matchingClient matchingservice.MatchingServiceClient,
+	historyRawClient resource.HistoryRawClient,
```

Comment:
Just wondering, why redefine `historyRawClient` here when it is already part of `transfer_queue_task_executor_base` ?

Reply:
oh that's because the existing history client in executor base is from shard context and is a retryable history client. But I need a raw history, so need to explicit pass it in. 

Another semi-related concern is that we've decided to stop using shard context as a property bag for passing dependencies around, at least not adding any new properties to it. This change also reduces the call on those GetXXX() methods on shard context by one :)

Reply:
It looks like you're also updating the history client in executor base to be a raw client in this PR though? https://github.com/temporalio/temporal/pull/4710/files#diff-46d3a4219811a59c9f3c620d18a5b4b5511913e0d1ae79c4be29229d5bd0d260R76

Reply:
Synced offline.